components/application-operator/pkg/controller/reconciler.go

```diff
@@ -77,35 +78,35 @@ func (r *applicationReconciler) handleErrorWhileGettingInstance(err error, reque
 	return reconcile.Result{}, logAndError(err, "Error getting %s Application", request.Name)
 }
 
-func (r *applicationReconciler) enforceDesiredState(application *v1alpha1.Application) error {
+func (r *applicationReconciler) enforceDesiredState(application *v1alpha1.Application) (func(application *v1alpha1.Application), error) {
```

Comment:
I would suggest creating a type alias for this function `func(application *v1alpha1.Application)`